pkg/engine/validation.go

```diff
@@ -29,12 +30,13 @@ import (
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/labels"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 // Validate applies validation rules from policy on the resource
-func Validate(ctx context.Context, rclient registryclient.Client, policyContext *PolicyContext) (resp *response.EngineResponse) {
+func Validate(ctx context.Context, rclient registryclient.Client, policyContext *PolicyContext, peLister kyvernov2alpha1.PolicyExceptionLister) (resp *response.EngineResponse) {
```

Comment:
Can we add the lister to the PolicyContext? This way we don't have to pass it around separately.

Reply:
IMHO we should use the same concept we used for configmap caching (a resolver interface), it will help supporting exceptions in the CLI if we want to.

Reply:
@eddycharly Is there any related link so that I can get some context?

Reply:
Search for ConfigMapResolver. Basically it’s a simple interface the engine depends on, it doesn’t care how the interface is implemented.

We should do the same, something like PolicyExceptionResolver taking a policy (and maybe a rule) and returns a list of PolicyException.

The way it is implemented doesn’t matter for now, it will probably use a lister behind the scene but it could also be a mock or read data from a file…

Reply:
I wonder if we can bake the autogen logic in the resolver or if we want to keep this explicit ?